### PR TITLE
Lint cleanup and minor style updates

### PR DESCRIPTION
**Pull request by russhwolf**
Style updates include applying the kotlin style guide. Not sure if we want to adjust our existing code style file on sherpa to include that, and possibly include any modifications we might want to make.

Will note a couple spots in comments that might merit discussion.

### CODE REVIEW

File: app/src/main/kotlin/io/intrepid/skotlinton/base/BaseActivity.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ abstract class BaseActivity : AppCompatActivity() {
     }
 
     @CallSuper
-    override fun onNewIntent(intent: Intent?) {
+    override fun onNewIntent(intent: Intent) {
```
Comment (russhwolf) [inline app/src/main/kotlin/io/intrepid/skotlinton/base/BaseActivity.kt]: Pretty confident this can't actually be null (how would the system know a null intent should be delivered to our app?) but let me know if I'm mistaken

File: app/src/main/kotlin/io/intrepid/skotlinton/base/BaseFragment.kt [hunk 2]
```diff
@@ -8,16 +8,15 @@ import android.support.v4.app.Fragment
...
 import butterknife.Unbinder
 import io.intrepid.skotlinton.SkotlintonApplication
 import timber.log.Timber
 
-abstract class BaseFragment<P : BaseContract.Presenter> : Fragment(), BaseContract.View {
+abstract class BaseFragment<out P : BaseContract.Presenter> : Fragment(), BaseContract.View {
 
     protected val skotlintonApplication: SkotlintonApplication
-        get() = activity?.application as SkotlintonApplication
+        get() = activity!!.application as SkotlintonApplication
```
Comment (russhwolf) [inline app/src/main/kotlin/io/intrepid/skotlinton/base/BaseFragment.kt]: @f2rf2r mentioned on my last PR that it might be nice to have some sort of `val a = activity!!` to wrap this, but I couldn't come up with a good name for it. Mostly made this change to flag for discussion, but we might also decide that this is the style we want to use in fragments moving forward so it might be worth leaving in as a demo.

Not also that the same thing applies for `context`.

File: app/build.gradle [hunk 3]
```diff
@@ -92,7 +92,7 @@ dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 
     implementation('com.crashlytics.sdk.android:crashlytics:2.8.0@aar') {
-        transitive = true;
+        transitive = true
```
Comment (f2rf2r) [inline app/build.gradle]: this got me thinking, is this flag even necessary? aren't dependencies transitive by default?

File: app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt [hunk 4]
```diff
@@ -16,8 +16,9 @@ class Example1Activity : BaseMvpActivity<Example1Contract.Presenter>(), Example1
         return Example1Presenter(this, configuration)
     }
 
+    @Suppress("ProtectedInFinal")
     @OnClick(R.id.example1_button)
-    internal fun onButtonClick() {
+    protected fun onButtonClick() {
```
Comment (russhwolf) [inline app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt]: I personally really like this pattern for consuming generated java code from kotlin, but I haven't seen it espoused elsewhere so I want to flag it for discussion here (it's also come up in slack a couple of times).

This method needs to be visible to Butterknife, but otherwise we don't want it exposed. The idiomatic pure-Kotlin way to do so would be to make it `internal`, and that's what the linter will push you toward. `protected` members in Kotlin are visible in Kotlin only to subclasses, and not to other code in the same package. However, Java code in the same package can access these things, and that's what Butterknife generates. Therefore making this `protected` hides this method from all of our other Kotlin code while still letting Butterknife do it's thing. The Kotlin linter thinks there's no reason to have a protected member in a final class because it doesn't consider this interop case, so we suppress the warning it generates.

The same pattern works for Dagger `@Inject` items, among other things.

**Comment by russhwolf** [comment]
One other note: I converted some functions to expression bodies, but only those that would be one-liners. This is the advice of the [android kotlin style guide](https://android.github.io/kotlin-guides/style.html) which is not part of the Jetbrains style guide and thus not accounted for in the default IDE settings. We may want to discuss our preferences there.

File: app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt (hunk 4 above)
Comment (f2rf2r) [reply app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt]: yeah, protected sounds better to me too. it's a shame kotlin doesn't support package protected, this is one cons for kotlin compared to java

Comment (shadow7) [reply app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt]: Lets remove butterknife and use synthetic accessors

File: app/build.gradle (hunk 3 above)
Comment (shadow7) [reply app/build.gradle]: I tried to compile once without the transitive statement and ran into issues. I agree with your logic though and maybe this was fixed in gradle 4.5.

File: app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt (hunk 4 above)
Comment (russhwolf) [reply app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt]: Unless we go the full Anko route, I still like butterknife for listeners, even if you want to use kotlinx.android for view references.

File: app/build.gradle (hunk 3 above)
Comment (f2rf2r) [reply app/build.gradle]: this has bugged me enough to finally google it https://stackoverflow.com/questions/44902339/whats-the-meaning-of-aar-with-transitive-true
TLDR: simply using `implementation 'com.crashlytics.sdk.android:crashlytics:2.8.0'` should work

Comment (russhwolf) [reply app/build.gradle]: Doing this causes the build to fail, due to problems resolving Crashlytics dependencies. I think there might be some weird stuff in the fabric gradle plugin that makes this a necessity.

Comment (russhwolf) [reply app/build.gradle]: oh wait I missed the piece on removing the `@aar`. That does seem to work.

**Comment by IntrepidBuildServer** [comment]
[42% (+42.34%) vs master 0%](https://ci.intrepid.io/job/Android/job/pull-requests/job/skotlinton-android-pr/36/)

**Comment by russhwolf** [comment]
Any other thoughts? I'd love to get this in before we start spinning up a bunch more kotlin work.
